examples/fizzbuzz/main.c

```diff
@@ -0,0 +1,189 @@
+    strcpy(fb->result, "");
+    fb->signalEmitter = signalEmitter;
+
+    return (qmlbind_client_object *) fb;
+}
+
+
+void fizzbuzz_delete_object_cb(qmlbind_client_object *obj)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+    free(fb->result);
+    qmlbind_signal_emitter_release(fb->signalEmitter);
+    free(fb);
+}
+
+
+qmlbind_value *fizzbuzz_call_method_cb(qmlbind_engine *engine,
+                                       qmlbind_client_object *obj,
+                                       const char *name,
+                                       int argc,
+                                       const qmlbind_value *const *argv)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "quit") == 0) {
+        exit(0); // FIXME: should call app.quit()
+    } else {
+        return qmlbind_value_new_undefined();
+    }
+}
+
+
+qmlbind_value *fizzbuzz_get_property_cb(qmlbind_engine *engine,
+                                        qmlbind_client_object *obj,
+                                        const char *name)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "input") == 0) {
+        return qmlbind_value_new_number(fb->input);
+    } else if (strcmp(name, "result") == 0) {
+        return qmlbind_value_new_string_cstr(fb->result);
+    } else {
+        return qmlbind_value_new_undefined();
+    }
+}
+
+
+void fizzbuzz_set_property_cb(qmlbind_engine *engine,
+                              qmlbind_client_object *obj,
+                              const char *name,
+                              const qmlbind_value *value)
+{
+    fizzbuzz_t *fb = (fizzbuzz_t *) obj;
+
+    if (strcmp(name, "input") == 0) {
+        fb->input = (int) qmlbind_value_get_number(value);
+
+        on_changed(fb, value);
+
+        const qmlbind_value *params[1] = {
+            qmlbind_value_new_number(fb->input)
+        };
+        qmlbind_signal_emitter_emit(fb->signalEmitter, "inputChanged", 1, params);
+        qmlbind_value_release(params[0]);
+    } else if (strcmp(name, "result") == 0) {
+        qmlbind_string *qstr = qmlbind_value_get_string(value);
+        const char *str = qmlbind_string_get_chars(qstr);
+        strcpy(fb->result, str);
+
+        const qmlbind_value *params[1] = {
+            qmlbind_value_new_string_cstr(fb->result)
+        };
+        qmlbind_signal_emitter_emit(fb->signalEmitter, "resultChanged", 1, params);
+        qmlbind_value_release(params[0]);
+    }
+}
+
+
+void create_fizzbuzz_metaclass()
+{
+    qmlbind_client_class *classRef = NULL;
+    qmlbind_client_callbacks callbacks;
+    callbacks.new_object = fizzbuzz_new_object_cb;
+    callbacks.delete_object = fizzbuzz_delete_object_cb;
+    callbacks.call_method = fizzbuzz_call_method_cb;
+    callbacks.get_property = fizzbuzz_get_property_cb;
+    callbacks.set_property = fizzbuzz_set_property_cb;
+
+    qmlbind_metaclass *metaclass =
+        qmlbind_metaclass_new(classRef, "FizzBuzz", callbacks);
+
+    char *params1[] = { "input" };
```

Comment:
`const char *params` and `const char *params2` make the const warnings go away, because then C is able to convert the constness automatically.